consensus/istanbul/backend/engine_test.go

```diff
@@ -1115,8 +1161,7 @@ func TestSnapshot_Validators_AddRemove(t *testing.T) {
 				}
 				if v.key == "governance.addvalidator" {
 					for _, i := range indices {
-						nodeKeys = append(nodeKeys, allNodeKeys[i])
-						addrs = append(addrs, allAddrs[i])
+						includeNode(allAddrs[i], allNodeKeys[i])
```

Comment:
Why are these lines changed?

Reply:
Without them, the same node ends up twice in `nodeKeys`/`addrs`, resulting in "invalid committed seals".

Reply:
@ian0371 What I meant to say was that changing the test code because of the core code change looks strange.
Do we need to sync the test code logic and the core code logic manually for this test?

Reply:
@ehnuje  Changing these lines is inevitable because the existing code cannot handle the new test cases. You can consider it "fixing a bug". 
I could add a new test function to verify them, but instead, I decided to reuse the existing function because most parts (i.e., except for these lines) will be reused. You might not want to go down that road.